zeppelin-web/src/app/notebook/notebook.controller.js

```diff
@@ -71,7 +71,9 @@ angular.module('zeppelinWebApp').controller('NotebookCtrl', function($scope, $ro
   $scope.runNote = function() {
     var result = confirm('Run all paragraphs?');
     if (result) {
-      $scope.$broadcast('runParagraph');
+      _.forEach($scope.note.paragraphs, function(n, key) {
+        angular.element('#' + n.id + '_paragraphColumn_main').scope().runParagraph(n.text);
```

Comment:
n.id is the paragraph id, so it has nothing to do with the order of paragraphs in the notebook.
$scope.note.paragraphs is an Array so paragraphs are already ordered, that's why we loop through it.